quartz_solar_forecast/inverters/givenergy.py

```diff
@@ -0,0 +1,48 @@
+import os
+import requests
+import pandas as pd
+from datetime import datetime
+from dotenv import load_dotenv
+
+# Load environment variables
+load_dotenv()
+
+def get_givenergy_latest_data():
+    """
+    Fetch the latest data from the GivEnergy API and return a DataFrame.
+    
+    :return: DataFrame with timestamp and power_kw columns
+    """
+    # Get API details from environment variables
+    api_key = os.getenv('GIVENERGY_API_KEY')
+    inverter_serial_number = os.getenv('GIVENERGY_INVERTER_SERIAL_NUMBER')
```

Comment:
At the moment this requires someone to know their inverter serial number? Could you add the logic to get the inverter serial number from the other inverter list endpoint rather than someone having to set it? Requires less manual steps for the user